### PR TITLE
update to TileDB-2.20.0

### PR DESCRIPTION
**Pull request by DimitrisStaratzis**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -104,7 +104,7 @@ add_library(tiledbjni SHARED
     src/main/c/custom/tiledb_custom.cxx
 )
 
-target_compile_definitions(tiledbjni PRIVATE -DTILEDB_CORE_OBJECTS_EXPORTS)
+target_compile_definitions(tiledbjni PRIVATE -Dtiledb_EXPORTS)
```
Comment (ihnorton) [inline CMakeLists.txt:107]: @DimitrisStaratzis do we actually need this?

@teo-tsirpanis why did this name change?

**Review by ihnorton — APPROVED** [review]
LGTM other than the cmake question.

File: CMakeLists.txt (hunk 1 above)
Comment (ihnorton) [reply CMakeLists.txt:107]: AFAICT the reason for this `-D` is to enable `TILEDB_EXPORT` in the https://github.com/TileDB-Inc/TileDB-Java/blob/master/swig/tiledb_java_extensions.h file. So it is effectively just piggybacking on the core exports.h (we could generate our own for -Java with cmake). So probably this is the ~only usage.

Comment (teo-tsirpanis) [reply CMakeLists.txt:107]: The define was renamed in TileDB-Inc/TileDB#4528. I wouldn't consider it a breaking change; this is an undocumented implementation detail that no external repository should define. This is in fact [the only use of this define](https://github.com/search?q=-DTILEDB_CORE_OBJECTS_EXPORTS+NOT+is%3Afork&type=code).

**Pull request merged**